xbuild/src/devices/adb.rs

```diff
@@ -11,8 +11,44 @@ use std::time::Duration;
 pub(crate) struct Adb(PathBuf);
 
 impl Adb {
-    pub fn which() -> Result<Self> {
-        Ok(Self(which::which(exe!("adb"))?))
+    pub fn which() -> Result<PathBuf> {
+        const ADB: &str = exe!("adb");
+
+        match which::which(ADB) {
```

Comment:
I thought for a second about using `which::which_in_global(ADB, paths)` but that requires `paths` to be a single concatenated string of `env("PATH"):env("ANDROID_HOME")` in the event that these env vars exist, and that overall looks more messy without including any context about `PATH` vs `ANDROID_HOME` vs `ANDROID_NDK_ROOT` providing this exe.